astroquery/exoplanet_archive/exoplanet_archive.py

```diff
@@ -0,0 +1,104 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import json
+import os
+from astropy.utils.data import download_file
+from astropy.io import ascii
+import astropy.units as u
+
+__all__ = ['ExoplanetArchive']
+
+EXOPLANETS_CSV_URL = ('http://exoplanetarchive.ipac.caltech.edu/cgi-bin/'
+                      'nstedAPI/nph-nstedAPI?table=exoplanets')
+TIME_ATTRS = {'rowupdate': 'iso'}
+BOOL_ATTRS = ('pl_kepflag', 'pl_ttvflag', 'pl_k2flag', 'st_massblend',
+              'st_optmagblend', 'st_radblend', 'st_teffblend')
+
+
+class ExoplanetArchiveClass(object):
+    """
+    Exoplanet Archive querying object. Use the ``get_table`` or ``get_planet``
+    methods to get information about exoplanets via the Exoplanet Archive.
+    """
+    def __init__(self):
+        self._param_units = None
+        self._table = None
+
+    @property
+    def param_units(self):
+        if self._param_units is None:
+            module_dir = os.path.dirname(os.path.abspath(__file__))
+            units_file = open(os.path.join(module_dir, 'data',
+                                           'exoplanet_nexsci_units.json'))
+            self._param_units = json.load(units_file)
+
+        return self._param_units
+
+    def get_confirmed_planets_table(self, cache=True, show_progress=True):
+        """
+        Download (and optionally cache) the `NExScI Exoplanet Archive Confirmed
+        Planets table <http://exoplanetarchive.ipac.caltech.edu/index.html>`_.
+
+        The Exoplanet Archive table returns lots of columns of data. A full
+        description of the columns can be found `here
+        <https://exoplanetarchive.ipac.caltech.edu/docs/API_exoplanet_columns.html>`_
+
+        Parameters
+        ----------
+        cache : bool (optional)
+            Cache exoplanet table to local astropy cache? Default is `True`.
+        show_progress : bool (optional)
+            Show progress of exoplanet table download (if no cached copy is
+            available). Default is `True`.
+        Returns
+        -------
+        table : `~astropy.table.Table`
+            Table of exoplanet properties.
+        """
+        if self._table is None:
+            table_path = download_file(EXOPLANETS_CSV_URL, cache=cache,
+                                       show_progress=show_progress, timeout=120)
+            exoplanets_table = ascii.read(table_path)
+
+            # Group by the host name
+            exoplanets_table.group_by('pl_hostname')
+
+            # Store column of lowercase names for indexing:
+            lowercase_names = [' '.join([host_name.lower(), letter])
```

Comment:
The space between the host name and the planet letter is often dropped, e.g. people often say "Trappist-1b" instead of "Trappist-1 b".  Let's drop the space here, which will make it easier to support both "Trappist-1b" and "Trappist-1 b" down in `query_planet`.

Reply:
Good idea!